web-ui/src/test/java/com/tutrit/controller/UserControllerTests.java

```diff
@@ -0,0 +1,89 @@
+package com.tutrit.controller;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.tutrit.bean.User;
+import com.tutrit.config.MySpringContext;
+import com.tutrit.gateway.UserGateway;
+import org.hamcrest.Matchers;
+import org.junit.jupiter.api.Test;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.autoconfigure.web.servlet.AutoConfigureMockMvc;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.boot.test.mock.mockito.MockBean;
+import org.springframework.test.web.servlet.MockMvc;
+import org.springframework.test.web.servlet.MvcResult;
+import org.springframework.test.web.servlet.request.MockMvcRequestBuilders;
+import org.springframework.test.web.servlet.result.MockMvcResultHandlers;
+import org.springframework.test.web.servlet.result.MockMvcResultMatchers;
+
+import java.util.Objects;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.model;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.view;
+
+@AutoConfigureMockMvc
+@SpringBootTest(classes = MySpringContext.SpringConfig.class)
+class UserControllerTests {
+    @MockBean
+    UserGateway userGateway;
+
+    @Autowired
+    MockMvc mockMvc;
+
+    @Autowired
+    ObjectMapper objectMapper;
+
+    @Test
+    void findUserById() throws Exception {
+        when(userGateway.findUserById("666"))
+                .thenReturn(createUser());
+        MvcResult result = mockMvc
+                .perform(MockMvcRequestBuilders.get("/users/666"))
+                .andDo(MockMvcResultHandlers.print())
+                .andExpect(MockMvcResultMatchers.status().isOk())
+                .andExpect(model().attribute("user", Matchers.equalTo(createUser())))
+                .andExpect(view().name("user-form.html"))
+                .andReturn();
+        User actualUser = (User) Objects.requireNonNull(result.getModelAndView()).getModel().get("user");
+        assertEquals(createUser(), actualUser);
+    }
+
+    @Test
+    void saveUser() throws Exception {
+        MvcResult result = mockMvc.perform(MockMvcRequestBuilders.post("/users/1?" +
+                                "userId=1" +
+                                "&name=FreddyMercury" +
+                                "&phoneNumber=+37525987")
+                        .param("save", "push"))
+                .andExpect(view().name("redirect:/users/1"))
+                .andExpect(MockMvcResultMatchers.status().is3xxRedirection())
+                .andDo(MockMvcResultHandlers.print())
+                .andReturn();
+        verify(userGateway).saveUser(verifyUser());
+    }
+
+    @Test
+    void deleteUser() throws Exception {
+        mockMvc.perform(MockMvcRequestBuilders.post("/users/1?" +
+                                "userId=1" +
```

Comment:
everything should go to 
.param(String, String)

Reply:
Done